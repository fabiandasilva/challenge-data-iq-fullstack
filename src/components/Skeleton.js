import React from "react";

const Skeleton = () => {
  return (
    <div className="mb-24 md:mb-0">
      <div className="block h-full rounded-lg bg-white shadow-[0_2px_15px_-3px_rgba(0,0,0,0.07),0_10px_20px_-2px_rgba(0,0,0,0.04)] animate-pulse">
        <div className="flex justify-center">
          <div className="flex justify-center -mt-[75px]">
            <div className="rounded-full bg-slate-700 h-[150px] w-[150px]"></div>
          </div>
        </div>
        <div className="px-6 my-1 pb-6 pt-0">
          <div className="h-5 w-20 bg-slate-700 rounded"></div>
          <div className="h-5 w-16 bg-slate-700 rounded mt-2"></div>
          <div className="mt-6 flex gap-1">
            <div className="h-5 w-5 bg-slate-700 rounded"></div>
            <div className="h-5 w-2/4 bg-slate-700 rounded"></div>
          </div>
          <div className="mt-3 flex gap-1">
            <div className="h-5 w-5 bg-slate-700 rounded"></div>
            <div className="h-5 w-2/4 bg-slate-700 rounded"></div>
          </div>
          <div className="mt-3 flex gap-1">
            <div className="h-5 w-5 bg-slate-700 rounded"></div>
            <div className="h-5 w-2/4 bg-slate-700 rounded"></div>
          </div>
          <div className="mt-3 flex gap-1">
            <div className="h-5 w-5 bg-slate-700 rounded"></div>
            <div className="h-5 w-2/4 bg-slate-700 rounded"></div>
          </div>
          <div className="h-9 w-24 bg-slate-700 rounded mx-auto mt-4"></div>
        </div>
      </div>
    </div>
  );
};

export default Skeleton;
