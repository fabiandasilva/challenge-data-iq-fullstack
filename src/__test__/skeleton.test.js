/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react";
import React from "react";
import Skeleton from "../components/Skeleton";

test("Skeleton se renderiza correctamente", () => {
  const { asFragment } = render(<Skeleton />);
  expect(asFragment()).toMatchSnapshot();
});
