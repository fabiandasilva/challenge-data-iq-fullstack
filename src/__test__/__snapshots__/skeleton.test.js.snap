// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton se renderiza correctamente 1`] = `
<DocumentFragment>
  <div
    class="mb-24 md:mb-0"
  >
    <div
      class="block h-full rounded-lg bg-white shadow-[0_2px_15px_-3px_rgba(0,0,0,0.07),0_10px_20px_-2px_rgba(0,0,0,0.04)] animate-pulse"
    >
      <div
        class="flex justify-center"
      >
        <div
          class="flex justify-center -mt-[75px]"
        >
          <div
            class="rounded-full bg-slate-700 h-[150px] w-[150px]"
          />
        </div>
      </div>
      <div
        class="px-6 my-1 pb-6 pt-0"
      >
        <div
          class="h-5 w-20 bg-slate-700 rounded"
        />
        <div
          class="h-5 w-16 bg-slate-700 rounded mt-2"
        />
        <div
          class="mt-6 flex gap-1"
        >
          <div
            class="h-5 w-5 bg-slate-700 rounded"
          />
          <div
            class="h-5 w-2/4 bg-slate-700 rounded"
          />
        </div>
        <div
          class="mt-3 flex gap-1"
        >
          <div
            class="h-5 w-5 bg-slate-700 rounded"
          />
          <div
            class="h-5 w-2/4 bg-slate-700 rounded"
          />
        </div>
        <div
          class="mt-3 flex gap-1"
        >
          <div
            class="h-5 w-5 bg-slate-700 rounded"
          />
          <div
            class="h-5 w-2/4 bg-slate-700 rounded"
          />
        </div>
        <div
          class="mt-3 flex gap-1"
        >
          <div
            class="h-5 w-5 bg-slate-700 rounded"
          />
          <div
            class="h-5 w-2/4 bg-slate-700 rounded"
          />
        </div>
        <div
          class="h-9 w-24 bg-slate-700 rounded mx-auto mt-4"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
